recurring.go

```diff
@@ -0,0 +1,39 @@
+package adyen
+
+// RecurringDetailsRequest structure to list all recurring payment associated to a shopperReference
+type RecurringDetailsRequest struct {
+	MerchantAccount  string `json:"merchantAccount"`
+	ShopperReference string `json:"shopperReference,omitempty"`
+	// Not mandatory
+	Recurring *Recurring `json:"recurring,omitempty"`
+}
+
+// RecurringDetailsResult structure to hold the RecurringDetails
+type RecurringDetailsResult struct {
+	CreationDate string `json:"creationDate"`
+	Details      []struct {
+		RecurringDetail struct {
```

Comment:
Maybe it's better to decouple RecurringDetail?
Card structure already exists, so maybe it could be used there as well?

Reply:
I'll decouple RecurringDetail.
For Card, the existing struct uses int for expiryMonth and expiryYear but JSon contains string values. So the unmarshalling process throws an errror : "json: cannot unmarshal string into Go struct field Card.expiryMonth of type int"

Below an example of Json returned by Adyen :
{"creationDate":"2017-09-27T16:30:23+02:00","details":[{"RecurringDetail":{"acquirer":"TestPmmAcquirer","acquirerAccount":"TestPmmAcquirerAccount","additionalData":{"cardBin":"411111"},"alias":"H167852639363479","aliasType":"Default","card":{"expiryMonth":"8","expiryYear":"2018","holderName":"test","number":"1111"},"contractTypes":["RECURRING"],"creationDate":"2017-09-27T16:30:23+02:00","firstPspReference":"8815065226234138","paymentMethodVariant":"visa","recurringDetailReference":"8315065226234554","variant":"visa"}},{"RecurringDetail":{"acquirer":"TestPmmAcquirer","acquirerAccount":"TestPmmAcquirerAccount","additionalData":{"cardBin":"497794"},"alias":"F314134953496447","aliasType":"Default","card":{"expiryMonth":"8","expiryYear":"2018","holderName":"test","number":"9497"},"contractTypes":["RECURRING"],"creationDate":"2017-09-27T16:58:08+02:00","firstPspReference":"8815065242885772","paymentMethodVariant":"visagold","recurringDetailReference":"8415065242885172","variant":"visa"}},{"RecurringDetail":{"acquirer":"TestPmmAcquirer","acquirerAccount":"TestPmmAcquirerAccount","additionalData":{"cardBin":"558555"},"alias":"A829629780526225","aliasType":"Default","card":{"expiryMonth":"8","expiryYear":"2018","holderName":"test","number":"5583"},"contractTypes":["RECURRING"],"creationDate":"2017-09-28T11:07:09+02:00","firstPspReference":"8515065896290058","paymentMethodVariant":"mc","recurringDetailReference":"8315065896294334","variant":"mc"}}],"shopperReference":"346","invalidOneclickContracts":"false"}

Reply:
than I guess it's better to update card structure to string type for month and year
https://docs.adyen.com/developers/api-reference/payments-api#card 
f.e 1 (January) should be 01 as shown on API reference